### PR TITLE
✨ Add Mattermost Alerting

### PR DESCRIPTION
**Pull request by ziedzaiem**
Hi !

I asked before to add support for insecure parameter to custom alerts so I can push notifications to a self hosted Mattermost instance, it worked well but It did not contain as information as in Slack notifications.

Mattermost and Slack use pretty much the same format, so I duplicated Slack alerting code and tweaked it for Mattermost (changed the emoji and added more information to attachements).

mattermost-alerts

Sorry for the type in the commit message (it's mattermost and not mettermost).

Best regards,

### CODE REVIEW

File: alerting/provider/mattermost/mattermost.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+		color = "#36A64F"
+	} else {
+		message = fmt.Sprintf("An alert for *%s* has been triggered due to having failed %d time(s) in a row", service.Name, alert.FailureThreshold)
+		color = "#DD0000"
+	}
+	var results string
+	for _, conditionResult := range result.ConditionResults {
+		var prefix string
+		if conditionResult.Success {
+			prefix = ":white_check_mark: "
```
Comment (TwinProduction) [inline alerting/provider/mattermost/mattermost.go:35]:
```suggestion
			prefix = ":white_check_mark:"
```

File: alerting/provider/mattermost/mattermost.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+						},
+						{
+						"title": "Condition results",
+						"value": "%s",
+						"short": false
+						}
+					]
+				}
+			]
+		}`, message, message, alert.Description, color, service.URL, results),
```
Comment (TwinProduction) [inline alerting/provider/mattermost/mattermost.go:70]:
```suggestion
		Body: fmt.Sprintf(`{
  "text": "",
  "username": "gatus",
  "icon_url": "https://raw.githubusercontent.com/TwinProduction/gatus/master/static/logo.png",
  "attachments": [
    {
      "title": ":rescue_worker_helmet: Gatus",
      "fallback": "Gatus - %s",
      "text": "%s:\n> %s",
      "short": false,
      "color": "%s",
      "fields": [
        {
          "title": "URL",
          "value": "%s",
          "short": false
        },
        {
          "title": "Condition results",
          "value": "%s",
          "short": false
        }
      ]
    }
  ]
}`, message, message, alert.Description, color, service.URL, results),
```

**Review by TwinProduction — CHANGES_REQUESTED** [review]
Very nice work!

I've never used Mattermost myself, but since you even provided a screenshot, I trust that you've tested it on your end and it works as expected. 

Once you've applied the two suggestions - assuming you agree with them - I'll merge it and submit an integration request to Mattermost so that Gatus can be part of Mattermost's integration directory.

Thank you for the contribution 👍

**Comment by ziedzaiem** [comment]
Hello, 

I applied the two sugguestions ;) I added an example docker-compose file with mattermost preview edition if you want to try it out. The code is working well on my production environment.

Regards,

**Review by TwinProduction — APPROVED** [review]

**Comment by TwinProduction** [comment]
Released in [v1.5.0](https://github.com/TwinProduction/gatus/releases/tag/v1.5.0)

I've also submitted an integration request to Mattermost.

**Pull request merged**